### PR TITLE
Provide more info in `SbangPathError` to aid CI debugging

### PR DESCRIPTION
**Pull request by tldahlgren**
GitLab instances have different rules for creating their build paths, which can create a lot of confusion when people encounter the `SbangPathError`.  It's easy for people to confuse or conflate the platform-specific `shebang_limit` with`config:install_tree:padded_length`.

This PR simply adds the length (and path) to the exception message to provide a quick way for people performing CI to debug their issue.  It includes a new unit test as well.

The fully qualified path for the `sbang` script is included since it can help people determine how paths are constructed for a given CI instance, for example.  Having this information has proven useful on at least one ECP CI instance.

For example, the error in the unit test is:

```
Install tree root is too long. Spack cannot patch shebang lines when script path length (138) exceeds limit (127).
   /tmp/dahlgren/pytest-of-dahlgren/pytest-0/test_install_sbang_too_long0/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee/bin/sbang
```

@shahzebsiddiqui

### CODE REVIEW

**Comment by haampie** [comment]
This is very useful!

I wonder if in some possible world this error can be turned into a warning... 127 characters is not a whole lot of characters

File: lib/spack/spack/test/sbang.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 import llnl.util.filesystem as fs
 
+import spack.config
```
Comment (alalazo) [inline lib/spack/spack/test/sbang.py]: Why is this import needed?

**Review by alalazo — COMMENTED** [review]
LGTM, just a minor question

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
It seems like tests are failing on macOS

**Comment by tldahlgren** [comment]
> It seems like tests are failing on macOS

Yeah.  My test code didn't match the test so I was off by a few characters PLUS I was using the wrong variable in the loop. ☹️  

Unfortunately, I am getting authorization errors trying to access patch coverage to see what it thinks I'm still missing.

**Comment by alalazo** [comment]
@tldahlgren Let me check. Today there was another case here https://github.com/spack/spack/pull/26560#issuecomment-936762372 of codecov posting low coverage when it was 100%

**Comment by alalazo** [comment]
Looking at [this](https://app.codecov.io/gh/spack/spack/blob/d81ba815d8e36e96e5a35bbaec24f073df662331/lib/spack/spack/hooks/sbang.py) this also seems 100% covered.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Thanks @tldahlgren
